<!DOCTYPE html>

<html lang="English" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Interpreting results &#8212; MetaDIVE 1.0.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=b08954a9" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=80cc54d1"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Detailed Conda environments used" href="Desc_envs.html" />
    <link rel="prev" title="Run setup" href="Run_setup.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="interpreting-results">
<h1>Interpreting results<a class="headerlink" href="#interpreting-results" title="Link to this heading">¶</a></h1>
<section id="results-folders">
<h2>Results folders<a class="headerlink" href="#results-folders" title="Link to this heading">¶</a></h2>
<p>MetaDIVE will generate slightly different results based on the modules chosen but the main results of all runs will be contained in a results folder titled <strong>Summarised_results</strong></p>
<p>This folder will contain several subfolders depending on the modules used. Here we will break down the results of each folder and ways to interpret them</p>
<a class="reference internal image-reference" href="_images/Summarised_results_folder.png"><img alt="MetaDIVE Summary results" class="align-center" src="_images/Summarised_results_folder.png" style="width: 400px;" />
</a>
<p>This main summary results folder is located in the pipeline directory after analysis is complete.</p>
<p>/MetaDIVE/pipeline/Summarised_results</p>
</section>
<section id="summary-results-combined-figures-and-tables-contigs">
<h2>Summary_results_combined_figures_and_tables_contigs<a class="headerlink" href="#summary-results-combined-figures-and-tables-contigs" title="Link to this heading">¶</a></h2>
<p>The following folder contains a breakdown of all summary graphs and tables from MetaDIVE.</p>
<a class="reference internal image-reference" href="_images/summary_results_and_tables.png"><img alt="MetaDIVE Summary results" class="align-center" src="_images/summary_results_and_tables.png" style="width: 400px;" />
</a>
<p>All figures in MetaDIVE are saved as both png and pdf files for easy reading.</p>
<p>The key files to be aware of are</p>
<section id="top-viral-hits-combined-raws-plus-contigs">
<h3>Top_viral_hits_combined_raws_plus_contigs<a class="headerlink" href="#top-viral-hits-combined-raws-plus-contigs" title="Link to this heading">¶</a></h3>
<p><strong>Top_viral_hits_combined_raws_plus_contigs</strong> (txt or png or pdf) are only present if Single read analysis module is run.
These files contain the most comprehensive summaries of all MetaDIVE analyses for viral assignments. These files</p>
<p>The txt file of this contains the following column headers</p>
<p><strong>reads_assigned_through_contigs</strong></p>
<p><strong>superkingdom, phylum class, order,family,genus,species,subspecies</strong> (Taxonomy of closest viral match)</p>
<p><strong>taxid</strong> (taxID of closest viral match)</p>
<p><strong>average_percent_ident, min_percent_ident, max_percent_ident</strong> (The average, min and max pairwise identity match of the contig in your sample to the closest matching reference)</p>
<p><strong>length</strong> (the average length of contigs from your sample which matched to the closest matching reference</p>
<p><strong>number_contigs_assigned</strong> (The number of contigs which assigned to this viral species)</p>
<p><strong>mean_complexity_of_contigs</strong> (The average compexity of the DNA sequence of the generated contig which provides insights into how redundant the sequence is)</p>
<p><strong>contigs_assigned</strong> (binary yes/no for whether contigs were assigned to this species at all, can be ignored)</p>
<p><strong>additional_raw_reads_assigned</strong> (binary yes/no for whether single reads were assigned to this species at all, can be ignored)</p>
<p><strong>number_of_raw_reads_assigned</strong> (how many single reads were assigned to this reference species)</p>
<p><strong>mean_complexity_of_raw_reads</strong> (the average compexity of the DNA sequence of the reads assigned which provides insights into how redundant the sequences are)</p>
<p><strong>mean_identity_of_raw_reads</strong> (the average pairwise identity match of the reads matching to the the closest matching reference)</p>
<p><strong>mean_aligned_length_of_raw_reads</strong> (The average alignment length of the reads to this reference viral species from BLASTn)</p>
<p><strong>total_reads_assigned</strong> (the total number of reads assigned through contigs [all which assigned to that viral species] and reads [only those which assigned through BLASTx and BLASTn] )</p>
<p><strong>classification_level_estimate</strong> (informal estimate of the likelihood of the viral species assigned being the same species as the reference, this is still a work in progress)</p>
<p><strong>confidence_of_assignment_as_virus</strong> (informal estimate of the likelihood of the viral species identified being in the data at all vs a false positive, this is still a work in progress)</p>
<p><strong>Sample</strong> (Your sample name)</p>
<p>From this results file you can see a comprehensive summary of what was detected and how likely what was assigned as the closest match is to be true.</p>
</section>
<section id="summarybarplot-contigs-bacterial-eukaryote-and-combined-samples-top-hits-to-bacterial-eukaryote">
<h3>Summarybarplot_contigs(bacterial,Eukaryote) and Combined_samples_top_hits_to(bacterial,Eukaryote)<a class="headerlink" href="#summarybarplot-contigs-bacterial-eukaryote-and-combined-samples-top-hits-to-bacterial-eukaryote" title="Link to this heading">¶</a></h3>
<p>These files and figures contain the top 10 identified bacterial and Eukaryote species (excluding host if host identification module was used).
These are similar but with fewer columns than the above example of Top_viral_hits_combined_raws_plus_contigs. Caution should be used when reading these tables as bacterial and Eukaryote assignments are prone to false positive assignments
through blastx.</p>
</section>
<section id="summarybarplot-bacterial-eukaryote-virus-families">
<h3>Summarybarplot_(bacterial,Eukaryote,virus)_families<a class="headerlink" href="#summarybarplot-bacterial-eukaryote-virus-families" title="Link to this heading">¶</a></h3>
<p>These files and figures are the like the above summarybarplot contigs results except with just single read results from Diamond BLASTx. This is done before Single Read analysis and so no quality
checks are completed yet. These files are only useful if you do not want to run the full Single Read analysis module but want a rough idea of what the rest of the reads are composed of.</p>
</section>
<section id="combined-samples-top-hits-to-viral-species">
<h3>Combined_samples_top_hits_to_viral_species<a class="headerlink" href="#combined-samples-top-hits-to-viral-species" title="Link to this heading">¶</a></h3>
<p>There are multiple similar tables here which serve slightly different functions.</p>
<p><strong>Combined_samples_top_hits_to_Viral_species.txt</strong> (pairwise count matrix of reads assigned to top 10 viruses across all samples from only contigs)</p>
<p><strong>Combined_samples_top_hits_to_Viral_species_long_table_format.txt</strong> (As above but instead of a matrix is in a longer format with additional information of pairwise identity of alignments and alignment lengths)</p>
<p><strong>Combined_samples_top_hits_to_Viral_species_blastn_false_positive_check.txt</strong> and <strong>Combined_samples_top_hits_to_Viral_species_long_table_format_blastn_false_positive_check.txt</strong> (like the above two tables, but with added information on comparsions between BLASTx and BLASTn results to
help tease apart unusual results.</p>
</section>
<section id="combined-summary-reads-filtering-txt">
<h3>Combined_summary_reads_filtering.txt<a class="headerlink" href="#combined-summary-reads-filtering-txt" title="Link to this heading">¶</a></h3>
<p>Summary of all the reads assignments as part of the core module e.g, how many reads passed QC. were assigned as CO1, to host, to SSU, to bacteria, Viruses etc.</p>
<p>paired text file to <strong>summary_all_reads_assigned_and_filtered.(png/pdf)</strong> which shows this information in figure format</p>
<p>Similar also to <strong>Combined_summary_contigs_assiging.txt</strong> which has the same information but focussed on contigs, contig sizes, contig numbers, numbers assigned through
BLASTx, BLASTn or unassigned.</p>
</section>
<section id="gather-summary-files-r-environment-and-gather-summary-files-r-environment-false-positive-blastncheck-rdata">
<h3>gather_summary_files_R_environment and gather_summary_files_R_environment_false_positive_blastncheck.Rdata<a class="headerlink" href="#gather-summary-files-r-environment-and-gather-summary-files-r-environment-false-positive-blastncheck-rdata" title="Link to this heading">¶</a></h3>
<p>These are the R environments which contain all data used to create these summary results tables and figures.</p>
<p>If you need to adjust the tables/figures you can easily load these documents into R and recreate the specific images you are after.</p>
</section>
</section>
<section id="summary-results-per-sample-contigs">
<h2>Summary_results_per_sample_contigs<a class="headerlink" href="#summary-results-per-sample-contigs" title="Link to this heading">¶</a></h2>
<p>This folder contains sample specific results for the core module. A lot of this data is unnecessary to view as it has all been summarised in the folder
<strong>Summary_results_combined_figures_and_tables_contigs</strong> but it also includes non summarised data including</p>
<p><strong>[Sample]_summarycontighits_assigned_assembly.txt</strong> which contains all individual contig assignments and statistics around the BLAST assignments</p>
<p>as well as larger summary files like <strong>[Sample]_top100Viralhits_contigs.txt</strong> which contains the top 100 assignments instead of the top 10 reported in some of the combined summary top assignment tables</p>
<p>This folder also has a breakdown of the top returned SSU/LSU/CO1 reads hits per sample <strong>[Sample]_top10returnedspecies_(SSU/LSU/CO1)</strong></p>
</section>
<section id="raw-assemblies">
<h2>raw_assemblies<a class="headerlink" href="#raw-assemblies" title="Link to this heading">¶</a></h2>
<p>This folder contains the generated assemblies for each sample. This can be useful if you want to do a deep dive, or you can take these contigs and put them into other
Analysis pipelines such as nf_core to allow for accurate bacterial identification and genome assembly without having to refilter and de-novo assemble raw data</p>
</section>
<section id="co1-lsu-ssu-results-folders">
<h2>CO1/LSU/SSU_results folders<a class="headerlink" href="#co1-lsu-ssu-results-folders" title="Link to this heading">¶</a></h2>
<p>When microbiome or host depletion modules are turned on three folders summarising the three marker genes are created.</p>
<p>within each of these folders are three folders</p>
<p><strong>blastn_contigs</strong> which contains the BLASTn results of assembled contigs of these markers.</p>
<p><strong>assembled_contigs</strong> which contains the assembled contigs of each marker gene.</p>
<p><strong>read_homologies</strong> which contains the results html figures of a Last common ancestor analysis used to identify what species are present.</p>
<a class="reference internal image-reference" href="_images/LCA_LSU_results.png"><img alt="MetaDIVE LSU_LCA_results_html" class="align-center" src="_images/LCA_LSU_results.png" style="width: 300px;" />
</a>
</section>
<section id="combined-raw-reads-and-contigs-viruses">
<h2>combined_raw_reads_and_contigs_viruses<a class="headerlink" href="#combined-raw-reads-and-contigs-viruses" title="Link to this heading">¶</a></h2>
<p>This folder contains two key results types split by sample name.</p>
<section id="sample-virusall-sums-csv-html">
<h3>[Sample]virusall_sums.(csv/html)<a class="headerlink" href="#sample-virusall-sums-csv-html" title="Link to this heading">¶</a></h3>
<p>These files are very similar to the <strong>Top_viral_hits_combined_raws_plus_contigs</strong> having the same columns and data however it is sample specific results instead of combined.</p>
<p>This data is also in an html format for easier manual interrogation of important samples. Here the user can type any variables into the search bar
to look for patterns or key viral groups of interest.</p>
<a class="reference internal image-reference" href="_images/Results_table_html.png"><img alt="MetaDIVE Results table html" class="align-center" src="_images/Results_table_html.png" style="width: 300px;" />
</a>
<p>If the diverged reads/contigs module is run, a more complete csv file titled</p>
<p><strong>[Sample]_virusall_sums_incl_diverged_reads.csv</strong> whill be available. This file contains an additional
column of data for additional diverged reads detected.</p>
</section>
<section id="individual-viruses-detected">
<h3>Individual viruses detected<a class="headerlink" href="#individual-viruses-detected" title="Link to this heading">¶</a></h3>
<p>For every virus identified in <strong>[Sample]_virusall_sums</strong> a folder is created for that specific virus.
in this folder will be all contigs and all reads assigned to that viral species in fasta and fastq format for quick manual inspection
in programs like geneious.</p>
</section>
</section>
<section id="metabat-binned-contigs">
<h2>Metabat_binned_contigs<a class="headerlink" href="#metabat-binned-contigs" title="Link to this heading">¶</a></h2>
<p>To be completed</p>
</section>
<section id="reference-guided-viral-genomes">
<h2>Reference_guided_viral_genomes<a class="headerlink" href="#reference-guided-viral-genomes" title="Link to this heading">¶</a></h2>
<p>To be completed</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">MetaDIVE</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Overview.html">MetaDIVE Overview</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Installation_guide.html">Installation guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="Run_setup.html">Run setup</a></li>
</ul>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Interpreting results</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#results-folders">Results folders</a></li>
<li class="toctree-l2"><a class="reference internal" href="#summary-results-combined-figures-and-tables-contigs">Summary_results_combined_figures_and_tables_contigs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#summary-results-per-sample-contigs">Summary_results_per_sample_contigs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#raw-assemblies">raw_assemblies</a></li>
<li class="toctree-l2"><a class="reference internal" href="#co1-lsu-ssu-results-folders">CO1/LSU/SSU_results folders</a></li>
<li class="toctree-l2"><a class="reference internal" href="#combined-raw-reads-and-contigs-viruses">combined_raw_reads_and_contigs_viruses</a></li>
<li class="toctree-l2"><a class="reference internal" href="#metabat-binned-contigs">Metabat_binned_contigs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#reference-guided-viral-genomes">Reference_guided_viral_genomes</a></li>
</ul>
</li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Desc_envs.html">Detailed Conda environments used</a></li>
<li class="toctree-l1"><a class="reference internal" href="Desc_databases.html">Detailed databases information</a></li>
<li class="toctree-l1"><a class="reference internal" href="Desc_scripts.html">Detailed description of scripts files</a></li>
<li class="toctree-l1"><a class="reference internal" href="Desc_rules.html">Detailed description of rules files</a></li>
<li class="toctree-l1"><a class="reference internal" href="Troubleshooting.html">Common issues and troubleshooting</a></li>
<li class="toctree-l1"><a class="reference internal" href="References.html">Software references</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="Run_setup.html" title="previous chapter">Run setup</a></li>
      <li>Next: <a href="Desc_envs.html" title="next chapter">Detailed Conda environments used</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025, James O'Dwyer.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.2.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="_sources/Results.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>